rbc/tests/test_omnisci_column_basic.py

```diff
@@ -853,3 +853,21 @@ def col_dtype_fn(x, m, y):
         query = f'select * from table(col_dtype_fn(cursor(select {col} from {table}), 1))'
         _, result = omnisci.sql_execute(query)
         assert list(result) == [(r,)] * 5
+
+
+def test_column_enumerate(omnisci):
+    from rbc.externals.omniscidb import set_output_row_size
+
+    @omnisci('int32(Column<int32>, OutputColumn<int32>)')
+    def col_enumerate(x, y):
+        sz = len(x)
+        set_output_row_size(sz)
+        for i, e in enumerate(x):
+            y[i] = e
+        return sz
+
+    _, result = omnisci.sql_execute(
+        f'select * from table(col_enumerate(cursor(select i4 from {omnisci.table_name})))')
+    _, expected_result = omnisci.sql_execute(
+        f'select i4 from {omnisci.table_name}')
```

Comment:
The row order of `select` is undefined and to ensure correct order (that `enumerate` needs), one should use
```
select rowid, i4 from {omnisci.table_name} order by rowid;
```

Reply:
Yes. Now the `expected_result` ought to be uniquely defined. The query
```
select * from table(col_enumerate(cursor(select i4 from {omnisci.table_name})))
```
is vulnerable to the same effect but it could be that table functions results are always ordered. If not, at some point we should see this test breaking, until then, let's leave it as it is.